src/operator/tensor/init_op.h

```diff
@@ -62,6 +134,53 @@ void FillCompute(const nnvm::NodeAttrs& attrs,
   });
 }
 
+
+template<typename xpu>
+void RangeCompute(const nnvm::NodeAttrs& attrs,
+                  const OpContext& ctx,
+                  const std::vector<TBlob>& inputs,
+                  const std::vector<OpReqType>& req,
+                  const std::vector<TBlob>& outputs) {
+  using namespace mshadow;
+  using namespace mshadow::expr;
+  Stream<xpu> *s = ctx.get_stream<xpu>();
+  const RangeParam& param = nnvm::get<RangeParam>(attrs.parsed);
+  MSHADOW_TYPE_SWITCH(outputs[0].type_flag_, DType, {
+    Tensor<xpu, 1, DType> out = outputs[0].FlatTo1D<xpu, DType>(s);
+    ASSIGN_DISPATCH(out, req[0], range<DType>(param.start.value(),
+                                              param.stop.value(),
+                                              param.step,
+                                              param.repeat));
+  });
+}
+
+
+inline bool RangeShape(const nnvm::NodeAttrs& attrs,
+                       std::vector<TShape> *in_attrs,
+                       std::vector<TShape> *out_attrs) {
+  const RangeParam& param = nnvm::get<RangeParam>(attrs.parsed);
+  CHECK_EQ(in_attrs->size(), 0);
+  CHECK_EQ(out_attrs->size(), 1);
+  CHECK_NE(param.step, 0)
+    << "Range does not support step=0, received " << param.step;
+  CHECK(param.repeat > 0)
+    << "Range only supports repeat > 0, received " << param.repeat;
+  if (param.step > 0) {
+    CHECK(param.start.value() < param.stop.value())
```

Comment:
this will raise error if start is none.
the way to use this is if(param.stop) { stop = param.stop.value() } else { stop = end}